File: 0217. Contains Duplicate/0217.cpp
// time complexity O(nlogn)
// space complexity O(1)

class Solution {
    public:
        bool containsDuplicate(vector<int>& nums) {
            // solution with sort
            sort(nums.begin(), nums.end()); // O(nlogn) time complexity
            
            for(int i=1; i<nums.size(); i++){
                if(nums[i-1]==nums[i]) return true;
            }
    
            return false;
        }
    };

File: 0217. Contains Duplicate/0217.java
// time complexity O(n)
// space complexity O(n)
class Solution {
    public boolean containsDuplicate(int[] nums) {
        // hashmap solution
        Map<Integer, Integer> check = new HashMap<>();
        
        for(int i=0; i<nums.length; i++){
            if(check.containsKey(nums[i])) return true;
            check.put(nums[i],1);
        }

        return false;
    }
}